src/app/api/[[...route]]/routes/sharehouses.ruote.ts

```diff
@@ -1,14 +1,64 @@
 import { Hono } from 'hono';
 
-export const app = new Hono();
+import prisma from '@/lib/prisma';
+import { auth } from '@/lib/auth';
+import type { IAssignedData } from '@/utils/assigned-data';
+
+const app = new Hono();
 
 export default app;
 
-app.get('/', (c) => {
-  return c.json({ message: 'Share houses route' });
-});
+app.get('/', async (c) => {
+  const session = await auth();
+
+  if (!session) {
+    return c.json({
+      error: 'Unauthorized',
+    });
+  }
+
+  const shareHouses = await prisma.landlord.findUnique({
+    where: { id: session.user.id },
+    include: {
+      shareHouses: {
+        include: {
+          assignmentSheet: {
+            select: {
+              assignedData: true,
+            },
+          },
+        },
+      },
+    },
+  });
```

Comment:
Could you add an if statement that checks whether the `shareHouses` is null, meaning if the account actually exists or not, to avoid unexpected behavior?

Reply:
I added the if statement.